cmd/lnmuxd/run.go

```diff
@@ -48,6 +48,50 @@ func runAction(c *cli.Context) error {
 		return err
 	}
 
+	// Initialise instrumentation server
+	instServer := initInstrumentationServer(cfg.InstrumentationAddress)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	group, ctx := errgroup.WithContext(ctx)
+	// Wait for all goroutine to finish.
+	// Return 'group.Wait()' error if no previous error was encountered.
+	defer func() {
```

Comment:
Looking at this now, I wonder if things can be made more readable by having a `runAction` that calls into an inner `run(group) error`? To avoid the named parameter and sophisticated defer? Maybe it is too much though, not sure.